ocs_ci/deployment/aws.py

```diff
@@ -134,18 +150,190 @@ def destroy_cluster(self, log_level="DEBUG"):
+                'route53_domain_name': config.ENV_DATA['base_domain'],
+                'vm_type_masters': config.ENV_DATA['master_instance_type'],
+                'vm_type_workers': config.ENV_DATA['worker_instance_type'],
+                'num_workers': "3"  # TODO: read in num_workers
+            }
+            for key, value in upi_env_vars.items():
+                os.environ[key] = value
+
+            # ensure environment variables have been set correctly
+            for key, value in upi_env_vars.items():
+                assert os.getenv(key) == value, f"{os.getenv(key)} != {value}"
+
+            # git clone repo from openshift-qe repo
+            clone_repo(
+                constants.OCP_QE_MISC_REPO, self.upi_repo_path
+            )
+
+            # create install-dir inside upi_script_path
+            relative_cluster_path = "../../../../../../"
+            os.symlink(
+                os.path.join(
+                    relative_cluster_path,
+                    os.path.basename(self.cluster_path.rstrip('/'))
+                ),
+                os.path.join(self.upi_script_path, "install-dir")
+            )
+
+            # NOT A CLEAN APPROACH: copy openshift-install and oc binary to
+            # script path because upi script expects it to be present in
+            # script dir
+            bindir = os.path.join(os.getcwd(), 'bin')
+            shutil.copy2(
+                os.path.join(bindir, 'openshift-install'),
+                self.upi_script_path,
+            )
+            shutil.copy2(
+                os.path.join(bindir, 'oc'), self.upi_script_path
+            )
+
+        def deploy(self, log_cli_level='DEBUG'):
+            """
+            Exact deployment will happen here
+
+            Args:
+                log_cli_level (str): openshift installer's log level
+                    (default: "DEBUG")
+            """
+            logger.info("Deploying OCP cluster")
+            logger.info(
+                f"Openshift-installer will be using loglevel:{log_cli_level}"
+            )
+
+            # Invoke upi_on_aws-install.sh
+            cidir = os.getcwd()
+            logger.info("Changing CWD")
+            try:
+                os.chdir(self.upi_script_path)
+            except OSError:
+                logger.exception(
+                    f"Failed to change CWD to {self.upi_script_path} "
                 )
-        except Exception:
-            logger.error(traceback.format_exc())
+            logger.info(f"CWD changed to {self.upi_script_path}")
+
+            with open("./upi_on_aws-install.sh", "r") as fd:
+                buf = fd.read()
+            data = buf.replace("openshift-qe-upi", "ocs-qe-upi")
+            with open("./upi_on_aws-install.sh", "w") as fd:
+                fd.write(data)
+
+            sys.path.append(self.upi_script_path)
+            full_path = os.getcwd()
+            proc = Popen(
+                [f'{full_path}/'
```

Comment:
cannot be this and next line be on one line?

Reply:
We should use `os.path.join` here too instead of an f-string. `[os.path.join(full_path, 'upi_on_aws-install.sh')]`

Reply:
Done